### PR TITLE
  Add inventory adapter/aspect commands

### PR DESCRIPTION
**Pull request by mandarjog**
list of adapters and aspects available to the mixer build.
Also prints default configurations.

```
mjog-macbookpro:mixer mjog$ bazel-bin/cmd/server/mixs inventory adapter
adapter istio/denyChecker: Denies every check request
Params:
error:
  code: 9
  message: ""
  details: []

adapter istio/genericListChecker: Checks whether a symbol is present in a list.
Params:
listentries: []

adapter istio/ipListChecker: Checks whether an IP address is present in an IP address list.
Params:
providerurl: http://localhost
refreshinterval: 60
ttl: 120

adapter istio/stdioAccessLogger: Writes structured access log entries to a standard I/O stream
Params:
logstream: 0

adapter istio/stdioLogger: Writes structured log entries to a standard I/O stream
Params:
logstream: 0```

Aspect
```
mjog-macbookpro:mixer mjog$ bazel-bin/cmd/server/mixs inventory aspect
aspect istio/denyChecker
Params:
{}

aspect istio/listChecker
Params:
blacklist: false
checkattribute: src.ip

aspect istio/logger
Params:
logname: istio_log
logentrydescriptornames: []
severityattribute: ""
timestampattribute: ""
timestampformat: 2006-01-02T15:04:05Z07:00
```

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/istio/mixer/209)
<!-- Reviewable:end -->

### CODE REVIEW

File: cmd/server/inventory.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
+// Copyright 2016 Google Inc.
```
Comment (geeknoid) [inline cmd/server/inventory.go]: 2017

File: pkg/adapterManager/registry.go [hunk 2]
```diff
@@ -34,7 +34,8 @@ type registry struct {
 // newRegistry returns a new Builder registry.
 func newRegistry(builders []adapter.RegisterFn) *registry {
 	r := &registry{buildersByName: make(map[string]adapter.Builder)}
-	for _, builder := range builders {
+	for idx, builder := range builders {
+		glog.V(2).Infof("Registering [%d] %v", idx, builder)
 		builder(r)
 	}
 	glog.V(2).Infof("Available Builders: %v", r.buildersByName)
```
Comment (geeknoid) [inline pkg/adapterManager/registry.go]: Is this redundant now given the new log statement above?

**Comment by mandarjog** [comment]
Done.

**Comment by douglas-reid** [comment]
Do we want to expose an API for getting this information?  Should we be thinking about mixer APIs beyond control-plane functionality?

**Comment by mandarjog** [comment]
yes, we do want to expose this thru an API. convert the proto configurations into their json schema (or similar) formats that can be processed by the UI.

There is an open issue for that piece of work.
At present this is simply a discovery tool as we are writing configuration files.

**Pull request merged**